modules/core/src/main/java/org/apache/synapse/api/ApiUtils.java

```diff
@@ -158,6 +161,52 @@ public static String getSubRequestPath(MessageContext synCtx) {
         return (String) synCtx.getProperty(RESTConstants.REST_SUB_REQUEST_PATH);
     }
 
+    public static boolean identifyApi(API api, MessageContext synCtx) {
+        if (synCtx.getProperty(RESTConstants.IDENTIFIED_API) == null) {
+            String path = ApiUtils.getFullRequestPath(synCtx);
+            if (null == synCtx.getProperty(RESTConstants.IS_PROMETHEUS_ENGAGED) &&
```

Comment:
It seems there is an issue with the original code here, what if PROMETEUS_ENGAGED is present and matchApiPath is false, we still have to return false; Let's fix that.

Reply:
Added the new condition block, let me know if the debug log message needs to changed or not.

Reply:
I still don't understand why we are checking Prometheus is enabled. Based on your logic we should be able to just get rid of the check. Shall we check the history we are checking this condition?